src/any_llm/providers/openai/custom.py

```
@@ -0,0 +1,49 @@
+    Prefer ``AnyLLM.create_openai_compatible(...)``, which reports the caller's chosen
+    name as the provider identity (via a per-name subclass) rather than masquerading as
+    ``openai``. Constructing this class directly reports the generic ``openai_compatible``
+    identity.
+    """
+
+    PROVIDER_NAME = "openai_compatible"
+    PROVIDER_DOCUMENTATION_URL = "https://platform.openai.com/docs/api-reference"
+    ENV_API_KEY_NAME = "OPENAI_COMPATIBLE_API_KEY"
+    ENV_API_BASE_NAME = "OPENAI_COMPATIBLE_API_BASE"
+
+    def __init__(self, api_base: str, api_key: str | None = None, **kwargs: Any) -> None:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)